.github/workflows/gh-linux_mac.yml

```diff
@@ -0,0 +1,360 @@
+            bootstrap_args="${bootstrap_args} --enable-sanitizer=address --enable-debug";
+          fi
+          if [[ "$TILEDB_CI_TSAN" == "ON" ]]; then
+            # Add thread sanitizer flag if necessary
+            bootstrap_args="${bootstrap_args} --enable-sanitizer=thread --enable-debug";
+          fi
+          if [[ "${{ matrix.TILEDB_SERIALIZATION }}" == "ON" ]]; then
+            # Add serialization flag if necessary
+            bootstrap_args="${bootstrap_args} --enable-serialization";
+          fi
+          if [[ "$TILEDB_FORCE_BUILD_DEPS" == "ON" ]]; then
+            # Add superbuild flag
+            bootstrap_args="${bootstrap_args} --force-build-all-deps";
+          fi
+
+          # name: 'Install dependencies'
+
+          mkdir -p $GITHUB_WORKSPACE/build
+          cd $GITHUB_WORKSPACE/build
+
+          # Configure and build TileDB
+          echo "Bootstrapping with '$bootstrap_args'"
+          $GITHUB_WORKSPACE/bootstrap $bootstrap_args
+
+          make -j4
+          make examples -j4
+          make -C tiledb install
+
+          #- run: |
+          cd $GITHUB_WORKSPACE/build
+          ls -la
+
+          if [[ ( ${{ runner.os }} == "Linux" && "${{ matrix.TILEDB_S3 }}" == "ON" ) ]]; then
+            # make sure docker is still running...
+            printenv
+            docker ps -a
+          fi
+
+          make -j4 -C tiledb tiledb_unit
+
+          if [[ "${{ matrix.TILEDB_CI_ASAN }}" == "ON" ]]; then
+            export ASAN_OPTIONS=detect_leaks=0 LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libasan.so.4
+          fi
+
+          if [[ ( "${{ runner.os }}" == "macOS" && "${{ matrix.TILEDB_GCS }}" == "ON" )]]; then
+            # GCS unit tests are temporarily unsupported on CI. Fake success with
+            # this echo.
+            echo ::set-output name=TILEDB_CI_SUCCESS::1
+          else
+            # Bypass Catch2 Framework stdout interception with awk on test output
+            make check | awk '/1: ::set-output/{sub(/.*1: /, ""); print; next} 1'
+          fi
+          # Kill the running Minio server, OSX only because Linux runs it within
+          # docker.
+          if [[ ( "${{ runner.os }}" == "macOS" && "${{ matrix.TILEDB_S3 }}" == "ON" ) ]]; then
+            kill -n 9 $MINIO_PID
+          fi
+
+          # Kill the running Azurite server
+          if [[ "${{ matrix.TILEDB_AZURE }}" == "ON" ]]; then
+            kill -n 9 $AZURITE_PID
+          fi
+
+          # Kill the running GCS emulator server Linux only because OSX does not
+          # run the emulator
+          if [[ ( "${{ runner.os }}" == "macOS" && "${{ matrix.TILEDB_GCS }}" == "ON" ) ]]; then
+            kill -9 $GCS_PID
+          fi
+
+#          # - bash: |
+#          if [[ "$TILEDB_GCS" != "ON" ]]; then
+#            pushd $BUILD_REPOSITORY_LOCALPATH/examples/cmake_project
+#            mkdir build && cd build
+#            cmake -DCMAKE_PREFIX_PATH=$BUILD_REPOSITORY_LOCALPATH/dist .. && make
```

Comment:
IIRC this section and the explicit calls to the executable are here to catch linkage errors, please don't get rid of them unless we are 100% sure it is tested somewhere else.